src/HydroShocktubeCMA/test_hydro_shocktube_cma.cpp

```diff
@@ -0,0 +1,298 @@
+		}
+
+		consVar(i, j, k, RadSystem<ShocktubeProblem>::gasEnergy_index) = P_R / (gamma - 1.);
+		consVar(i, j, k, RadSystem<ShocktubeProblem>::gasInternalEnergy_index) = P_R / (gamma - 1.);
+		consVar(i, j, k, RadSystem<ShocktubeProblem>::gasDensity_index) = rho_R;
+		consVar(i, j, k, RadSystem<ShocktubeProblem>::x1GasMomentum_index) = 0.;
+		consVar(i, j, k, RadSystem<ShocktubeProblem>::x2GasMomentum_index) = 0.;
+		consVar(i, j, k, RadSystem<ShocktubeProblem>::x3GasMomentum_index) = 0.;
+
+		consVar(i, j, k, HydroSystem<ShocktubeProblem>::scalar0_index + 0) = 0.1 * rho_R;
+		consVar(i, j, k, HydroSystem<ShocktubeProblem>::scalar0_index + 1) = 0.3 * pow(sin(20 * 3.14 * 1), 2) * rho_R;
+		consVar(i, j, k, HydroSystem<ShocktubeProblem>::scalar0_index + 2) = 1 - 0.1 - 0.3 * pow(sin(20 * 3.14 * 1), 2) * rho_R;
+	}
+}
+
+template <> void RadhydroSimulation<ShocktubeProblem>::ErrorEst(int lev, amrex::TagBoxArray &tags, Real /*time*/, int /*ngrow*/)
+{
+	// tag cells for refinement
+
+	const Real eta_threshold = 0.1; // gradient refinement threshold
+	const Real rho_min = 0.01;	// minimum rho for refinement
+	auto const &dx = geom[lev].CellSizeArray();
+
+	for (amrex::MFIter mfi(state_new_cc_[lev]); mfi.isValid(); ++mfi) {
+		const amrex::Box &box = mfi.validbox();
+		const auto state = state_new_cc_[lev].const_array(mfi);
+		const auto tag = tags.array(mfi);
+
+		amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+			int const n = 0;
+			Real const rho = state(i, j, k, n);
+			Real const del_x = (state(i + 1, j, k, n) - state(i - 1, j, k, n)) / (2.0 * dx[0]);
+			Real const gradient_indicator = std::sqrt(del_x * del_x) / rho;
+
+			if (gradient_indicator > eta_threshold && rho >= rho_min) {
+				tag(i, j, k) = amrex::TagBox::SET;
+			}
+		});
+	}
+}
+
+template <> void RadhydroSimulation<ShocktubeProblem>::computeAfterTimestep()
+{
+	auto [position, values] = fextract(state_new_cc_[0], Geom(0), 0, 0.5);
+	const int nx = static_cast<int>(position.size()); // number of cells along the x direction
+	const int nmscalars = Physics_Traits<ShocktubeProblem>::numMassScalars;
+
+	if (amrex::ParallelDescriptor::IOProcessor()) {
+		userData_.t_vec_.push_back(tNew_[0]);
+		amrex::Real sum_Delta_eps_t = 0.0; // so we can plot the average error as a function of time
+
+		for (int nn = 0; nn < nx; ++nn) {
+			amrex::Real specieSum = 0.0;
+			amrex::Real Delta_eps_t = 1e100;
+
+			const amrex::Real rho = values.at(RadSystem<ShocktubeProblem>::gasDensity_index)[nn];
+
+			for (int n = 0; n < nmscalars; ++n) {
+				specieSum += values.at(HydroSystem<ShocktubeProblem>::scalar0_index + n)[nn];
+			}
+
+			Delta_eps_t = 1e0 - specieSum / rho; // normalize by density to convert partial density to mass fraction
+
+			if ((std::abs(Delta_eps_t) > 1.0e-13) || std::isnan(Delta_eps_t)) {
+				amrex::Print() << "Mass scalars not conserved to machine precision!\n";
+				consv_test_passes = false;
+			} else {
+				consv_test_passes = true;
```

Comment:
This is logically incorrect. If the last element passes, then it will pass, which is not what you want.